File: leetcode/python/0190-reverse-bits.py
# https://leetcode.com/problems/reverse-bits/
# Easy

# Continusly check for (n & 1) and append that to the back of ret
class Solution:
    def reverseBits(self, n: int) -> int:
        ret = 0
        for _ in range(32):
            ret = (ret << 1) | (n & 1)
            n >>= 1
        return ret

File: leetcode/c++/0190-reverse-bits.cc
// https://leetcode.com/problems/reverse-bits/
// Easy

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t ret = 0;
        for (int i = 0; i < 32; ++i, n >>= 1) {
            ret = (ret << 1) | (n & 1);
        }
        return ret;
    }
};